server/index.ts

```diff
@@ -1,13 +1,132 @@
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Authorization: `OAuth oauth_consumer_key="${consumerKey}",oauth_nonce="${Date.now()}",oauth_token="${oAuthRequestToken}", oauth_signature=${consumerSecret}&${oAuthRequestTokenSecret}",oauth_signature_method="PLAINTEXT",oauth_timestamp="${Date.now()}",oauth_verifier="${oAuthVerifier}"`,
+      },
+    }
+  );
+
+  const token = await tokenResponse.text();
+  const params = new URLSearchParams(token);
+  oAuthAccessToken = params.get('oauth_token');
+  oAuthAccessTokenSecret = params.get('oauth_token_secret');
+
+  const identityResponse = await fetch(
+    'https://api.discogs.com/oauth/identity',
+    {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Authorization: `OAuth oauth_consumer_key="${consumerKey}", oauth_nonce="${Date.now()}", oauth_token="${oAuthAccessToken}", oauth_signature="${consumerSecret}&${oAuthAccessTokenSecret}",oauth_signature_method="PLAINTEXT",oauth_timestamp="${Date.now()}"`,
+      },
+    }
+  );
+  const identity = await identityResponse.json();
+  const username = identity.username;
+  if (oAuthAccessToken && oAuthAccessTokenSecret) {
+    response.cookie('username', username, { httpOnly: true });
+    response.cookie('accessToken', oAuthAccessToken, { httpOnly: true });
+    response.cookie('accessTokenSecret', oAuthAccessTokenSecret, {
+      httpOnly: true,
+    });
+    response.redirect('http://localhost:3001/api/me');
+  }
+});
+
+app.get('/api/search/:searchq', async (request, response) => {
+  const searchQuery = request.params.searchq;
+
+  const token = request.cookies.accessToken;
+  const secret = request.cookies.accessTokenSecret;
+
+  const searchResponse = await fetch(
+    `https://api.discogs.com/database/search?q=${searchQuery}`,
+    {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Authorization: `OAuth oauth_consumer_key="${consumerKey}", oauth_nonce="${Date.now()}", oauth_token="${token}", oauth_signature="${consumerSecret}&${secret}",oauth_signature_method="PLAINTEXT",oauth_timestamp="${Date.now()}"`,
+      },
+    }
+  );
+  const search = await searchResponse.json();
+
+  response.send(search);
+});
+
+app.get('/api/me', async (request, response) => {
+  const user = request.cookies.username;
+  const token = request.cookies.accessToken;
+  const secret = request.cookies.accessTokenSecret;
+
+  const searchResponse = await fetch(
+    `https://api.discogs.com/users/${user}/collection`,
+    {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        Authorization: `OAuth oauth_consumer_key="${consumerKey}", oauth_nonce="${Date.now()}", oauth_token="${token}", oauth_signature="${consumerSecret}&${secret}",oauth_signature_method="PLAINTEXT",oauth_timestamp="${Date.now()}"`,
+      },
+    }
+  );
+  const search = await searchResponse.json();
+
+  response.send(search);
+});
```

Comment:
Good job!
Error handling in all of your endpoints would be great if the request is failing:
https://expressjs.com/en/guide/error-handling.html

```suggestion
app.get('/api/me', async (request, response, next) => {
  try {
    const user = request.cookies.username;
    const token = request.cookies.accessToken;
    const secret = request.cookies.accessTokenSecret;
  
    const searchResponse = await fetch(
      `https://api.discogs.com/users/${user}/collection`,
      {
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded',
          Authorization: `OAuth oauth_consumer_key="${consumerKey}", oauth_nonce="${Date.now()}", oauth_token="${token}", oauth_signature="${consumerSecret}&${secret}",oauth_signature_method="PLAINTEXT",oauth_timestamp="${Date.now()}"`,
        },
      }
    );
    const search = await searchResponse.json();
  
    response.send(search);
  } catch(error) {
    // Maybe create custom error response or use `next` for Express default error handler
    next(error);
  }
});
```